otlp/traces_test.go

```diff
@@ -1035,20 +717,13 @@ func TestEmptyOrInvalidServiceNameAttributeUsesDefault(t *testing.T) {
 				}},
 			}
 
-			bodyBytes, err := proto.Marshal(req)
-			assert.Nil(t, err)
-
-			buf := new(bytes.Buffer)
-			buf.Write(bodyBytes)
-
-			body := io.NopCloser(strings.NewReader(buf.String()))
 			ri := RequestInfo{
 				ApiKey:      "abc123DEF456ghi789jklm",
 				Dataset:     "legacy-dataset",
 				ContentType: "application/protobuf",
 			}
 
-			result, err := TranslateTraceRequestFromReader(body, ri)
+			result, err := TranslateTraceRequest(req, ri)
```

Comment:
Because this tests is less about decoding and unmarshaling the request (well-covered in the HTTP test cases) and more about what to do given request headers and the state of the telemetry, I opted to snip out the marshal/unmarshal and go direct to our function for translating an OTLP proto struct.

A few tests were trimmed down along these lines.